### PR TITLE
Add host_id filter for tls_certificates

### PR DESCRIPTION
**Pull request by timopollmeier**
**Checklist**:

- Tests N/A
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/manage_sql_tls_certificates.c [hunk 1]
```diff
@@ -192,12 +192,41 @@ gchar *
...
+          "     WHERE host = (SELECT id FROM hosts"
+          "                   WHERE uuid='%s')"
+          "       AND name = 'ip')"
+          "  SELECT tls_certificate"
+          "    FROM tls_certificate_sources AS sources"
+          "    JOIN tls_certificate_origins AS origins"
+          "      ON origins.id = sources.origin"
+          "    JOIN tls_certificate_locations AS locations"
+          "      ON locations.id = sources.location"
+          "    WHERE origins.origin_id"
```
Comment (mattmundell) [inline src/manage_sql_tls_certificates.c]: I'm struggling to get my head around this part.  It seem to me like you're selecting all tls_certificates that have any of the reports in host_idents as origin and at the same time any of the ip's in host_identifiers as location.  Doesn't that mean you will select a cert that report_a found at ip_2 and report_b found at ip_1, when the host_identifier was ip_1 by report_a?

Comment (timopollmeier) [reply src/manage_sql_tls_certificates.c]: You're right, the conditions for the report id and the IP address need to apply to the same host_idents entry simultaneously.

Comment (timopollmeier) [reply src/manage_sql_tls_certificates.c]: I've replaced the WHERE clause with another JOIN that has to match both the IP and the report id at the same time in 3c1ab3bae11b514110fe699d286701ee3f778aa3.

Comment (mattmundell) [reply src/manage_sql_tls_certificates.c]: Ya, makes more sense to me now.

**Comment by timopollmeier** [comment]
> Does the new filter term need to be added to the GMP docs?

Thanks for reminding me. I've added it in abb8828004aec2fe030ff6825143f19ee6802730.

**Pull request merged**